pkg/cmd/provisioning/utils.go

```diff
@@ -173,9 +173,21 @@ func GetListOfCredentialsRequests(dir string) ([]*credreqv1.CredentialsRequest,
 			// infrastructure.
 			if value, ok := cr.Annotations["release.openshift.io/delete"]; ok && value == "true" {
 				log.Printf("Ignoring CredentialsRequest %s/%s as it is marked for in-cluster deletion", cr.Namespace, cr.Name)
-			} else {
-				credRequests = append(credRequests, cr)
+				continue
 			}
+
+			// Ingore CredentialsRequest with the feature-gate annotation
+			if value, ok := cr.Annotations[featureGateAnnotation]; ok {
+				// By edict, the only valid value is "TechPreviewNoUpgrade"
+				if value != validTechPreviewAnnotationValue {
+					log.Printf("CredentialsRequests %s/%s has unexpected feature-gate value", cr.Namespace, cr.Name)
+
+				}
+				log.Printf("Ignoring CredentialsRequest %s/%s with tech-preview annotation", cr.Namespace, cr.Name)
```

Comment:
This will make it impossible for users to bootstrap a cluster as a TechPreviewNoUpgrade cluster on certain platforms right? I think this will break a number of release informing jobs and QE scenarios as we, as an org, decided a while back, bootstrapping as a TPNU cluster should be supported.

Reply:
We can just take a new CLI parameter to opt into processing these annotated manifests.